src/EditorFeatures/Core/Implementation/EditAndContinue/EditAndContinueDiagnosticUpdateSource.cs

```diff
@@ -110,5 +110,41 @@ private void RaiseDiagnosticsUpdated(DiagnosticsUpdatedArgs args)
                 updated(this, args);
             }
         }
+
+        private class EnCId : ErrorSourceId
+        {
+            private DebuggingSession _session;
+            private object _errorId;
+
+            public EnCId(DebuggingSession session, object errorId)
+            {
+                _session = session;
+                _errorId = errorId;
+            }
+
+            public override string ErrorSource
+            {
+                get
+                {
+                    return "Edit And Continue";
```

Comment:
Perhaps create WellKnownErrorSources and use WellKnownErrorSources.EditAndContinue, WellKnownErrorSources.Build etc.?